src/main/java/vi/viredis/pool/RedisPool.java

```diff
@@ -0,0 +1,127 @@
+package vi.viredis.pool;
+
+import vi.viredis.client.ViRedis;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedTransferQueue;
+import java.util.concurrent.TimeUnit;
+
+/*
+ *  implements the method of common pool interface
+ */
+public class RedisPool implements Pool {
+    public boolean isShutdown;
```

Comment:
variable 'isShutdown' is never updated. Make it private. Update it's value during shutdown() method call.

Reply:
done!